dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java

```diff
@@ -68,31 +74,62 @@ final class ConjureBodySerDe implements BodySerDe {
      */
     ConjureBodySerDe(
             List<WeightedEncoding> rawEncodings,
-            ErrorDecoder errorDecoder,
             EmptyContainerDeserializer emptyContainerDeserializer,
             CaffeineSpec cacheSpec) {
         List<WeightedEncoding> encodings = decorateEncodings(rawEncodings);
         this.encodingsSortedByWeight = sortByWeight(encodings);
         Preconditions.checkArgument(encodings.size() > 0, "At least one Encoding is required");
         this.defaultEncoding = encodings.get(0).encoding();
-        this.binaryInputStreamDeserializer = new EncodingDeserializerRegistry<>(
-                ImmutableList.of(BinaryEncoding.INSTANCE),
-                errorDecoder,
+        this.binaryInputStreamDeserializer = deserializerFor(
+                DeserializerType.STANDARD, emptyContainerDeserializer, BinaryEncoding.MARKER, Collections.emptyMap());
+        this.optionalBinaryInputStreamDeserializer = deserializerFor(
+                DeserializerType.STANDARD,
```

Comment:
I think these ought to be 
```suggestion
        this.binaryInputStreamDeserializer = deserializerFor(
                DeserializerType. INPUT_STREAM_OR_OPTIONAL_INPUT_STREAM, emptyContainerDeserializer, BinaryEncoding.MARKER, Collections.emptyMap());
        this.optionalBinaryInputStreamDeserializer = deserializerFor(
                DeserializerType. INPUT_STREAM_OR_OPTIONAL_INPUT_STREAM,
```
?

Reply:
Fixed!